source/Engine.cpp

```diff
@@ -2093,16 +2099,21 @@ void Engine::DoWeather(Weather &weather)
 	if(weather.HasWeapon() && !Random::Int(weather.Period()))
 	{
 		const Hazard *hazard = weather.GetHazard();
-		double multiplier = weather.DamageMultiplier();
 
 		// Get all ship bodies that are touching a ring defined by the hazard's min
 		// and max ranges at the hazard's origin. Any ship touching this ring takes
 		// hazard damage.
 		for(Body *body : shipCollisions.Ring(weather.Origin(), hazard->MinRange(), hazard->MaxRange()))
 		{
 			Ship *hit = reinterpret_cast<Ship *>(body);
-			double distanceTraveled = weather.Origin().Distance(hit->Position()) - hit->GetMask().Radius();
-			hit->TakeDamage(visuals, *hazard, multiplier, distanceTraveled, weather.Origin(), nullptr, hazard->BlastRadius() > 0.);
+			double multiplier = weather.DamageMultiplier();
```

Comment:
Note that this function call isn't a simple getter. If the hazard deviates (which the hazards in the Ember Waste do), it's doing calculations to compute the current multiplier. So moving the multiplier into the for loop means we're redoing those calculations for every ship.

Reply:
Right, this should indeed stay out of the loop then.